src/test/resources/org/broadinstitute/barclay/help/templates/BashTestDoclet/bash-completion.template.ftl

```diff
@@ -0,0 +1,343 @@
+
+####################
+# Tab completion file to allow for easy use of this tool with the command-line (bash/zsh)
+####################
+
+<#include "bash-completion.macros.ftl"/>
+
+####################################################################################################
+
+# High-level caller/dispatch script information:
```

Comment:
Having generic tab completion capability in Barclay would definitely be a bonus, but I'm not sure what it would take get there. Given that the test in this PR is fairly GATK-specific and isn't really testing Barclay functionality, I'm thinking it should live in GATK. @jonn-smith What do you think about, as a first step, separating out the core Barclay changes for this PR, and moving the rest (the .sh, .ftl, and test) to a PR in the GATK repo ?